### PR TITLE
Fix TestCtlV3Lock/Elect flakes

### PR DESCRIPTION
**Pull request by tjungblu**
waiting for a proper exit after a signal should ensure the goroutines don't leak.

Signed-off-by: Thomas Jungblut <tjungblu@redhat.com>

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by tjungblu** [comment]
the grpc proxy one seems unrelated, referring to the watch cache. Could you please re-run the e2e and the grpc tests a couple times to ensure it doesn't flake? I'm unable to trigger :/

**Comment by serathius** [comment]
First pass e2e tests passed, rerunning to see if they pass the second time.

File: tests/e2e/ctl_v3_lock_test.go [hunk 1]
```diff
@@ -69,7 +69,12 @@ func testLock(cx ctlCtx) {
 	if err != nil {
 		cx.t.Fatal(err)
 	}
-	defer blockAcquire.Stop()
+	defer func(blockAcquire *expect.ExpectProcess) {
```
Comment (serathius) [inline tests/e2e/ctl_v3_lock_test.go:72]: Should this be in defer? Would putting it after select that reads from `ch` enough?

Comment (tjungblu) [reply tests/e2e/ctl_v3_lock_test.go]: true, it makes more sense to not defer stop here

Comment (tjungblu) [reply tests/e2e/ctl_v3_lock_test.go]: hmm, seems like there was some reason after all:
> 2022-11-17T13:44:28.5305567Z     ctl_v3_lock_test.go:106: expected different lock name, got l1="a/186d8485d2349505\r\n", l2="Error: context canceled\r\n"

**Pull request merged**